### PR TITLE
Adds an ert: Captain Dropkick!

### PR DESCRIPTION
**Pull request by tralezab**
:cl:
add: Captain Dropkick is a new ERT!
/:cl:

I'm gonna see if this doesn't get closed before i make sprites but I imagine some kind of shitty superhero suit or something like that

the dropkicks themselves are very lethal and give you time to get up after the dropkick, but missing and hitting objects can hurt a lot and even just missing and hitting the floor means you have to get up.

don't hit a wall

Also i'm not entirely happy with the name "Captain Dropkick" if anyone has a better idea i'll probably replace it with that

### CODE REVIEW

**Comment by HomocidlePancakes** [comment]
We getting our own Mattchoe Meen

**Comment by OverlyGeneric** [comment]
I have several ideas for this.

I think the names **Jumpin' Joey**, references the marsupial and the Australian wrestler Joe Savoldi who first used the dropkick, or **The Human Missle.**

I think this "ERT" should have a kickboxing martial art, maybe a lesser form of Sleepy Carp, that allows him to do moves like leg sweeps, roundhouse kicks, and dropkicks. 

Additionally, I think this would be _hilarious_ to add an attack that targets the groin and stuns male players for a few seconds. Female characters should still get stunned but albeit much shorter as it's painful for women to get kicked into the groin but not as "brutal."

If I were to imagine what the "ERT" would look like more of a boxer than what other ERTs are like. Muscular thighs maybe?

**Comment by SouDescolado** [comment]
Just to make sure, is it ok that the chat message comes before the confirmation that it worked?

For example, if there is a book that you read and unlearn "Dropkick", but you never knew it, you still get the message it got removed, right?

**Comment by tralezab** [comment]
I'm not sure what you are asking, when you wear captain dropkick's boots you learn the power to dropkick, when you take them off you unlearn it

**Comment by ExcessiveUseOfCobblestone** [comment]
did we remove the wrestling belt? how sad!

File: code/datums/martial/dropkicking.dm [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+/datum/martial_art/dropkick/throw_impact_act(atom/hit_atom, datum/thrownthing/throwingdatum)
+	//
+	if(!dropkicking)
+		return//this will call the rest of throw_impact.
+	//owner.Paralyze(20)//you still always take longer to get up because it's a dropkick
+	if(iscarbon(hit_atom) && hit_atom != src)
+		var/mob/living/carbon/victim = hit_atom
+		if(victim.movement_type & FLYING)
+			return FALSE//this does not call the rest of throw_impact.
```
Comment (JJRcop) [inline code/datums/martial/dropkicking.dm]: Doesn't this fail the if check here, causing the throw_impact to continue?
https://github.com/tgstation/tgstation/blob/cd4b3c4b4fdb0b90eea7f41ae5b4804c08b684ce/code/modules/mob/living/carbon/carbon.dm#L98-L100

Did this work in testing?

File: code/datums/martial.dm [hunk 2]
```diff
@@ -21,6 +21,9 @@
 /datum/martial_art/proc/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
 	return 0
 
+/datum/martial_art/proc/throw_impact_act(atom/hit_atom, datum/thrownthing/throwingdatum)
+	return//if you do not want this to continue on to throw_impact, then return anything not null (see dropkick for an example)
```
Comment (JJRcop) [inline code/datums/martial.dm]: Don't if conditions fail on any falsey value, like `null`, `""`, `0`, and `FALSE` (which is 0)?

Comment (tralezab) [reply code/datums/martial.dm]: better way to word that thanks

File: code/datums/martial/dropkicking.dm (hunk 1 above)
Comment (tralezab) [reply code/datums/martial/dropkicking.dm]: it absolutely does not i should probably go with
if(!martial_return)
instead so the returns make more sense

**Comment by tralezab** [comment]
Let me resolve problems with throw_impact_act, and then i'm going to cautiously start coding the captain dropkick suit...

**Comment by OverlyGeneric** [comment]
> I want to make sure the captain sticks to dropkicks as it is the most risky part of his kit, I was probably going to prevent him from attacking people not on the ground because "I'm captain dropkick, why punch when I can DROPKICK?"

I know it's his gimmick and all but its going to get boring pretty quickly if that's the only thing he can do.

It's his riskiest move but shouldn't be his only move that he knows.

**Comment by JJRcop** [comment]
Every time he drops an item he can't help but kick it.

**Comment by tralezab** [comment]
Oh fuck yeah I'll do that

**Comment by JJRcop** [comment]
To clarify, the last two examples are just to show how RegisterSignal and UnregisterSignal are used properly by a non-component datum. Don't use `COMSIG_MOVABLE_HEAR`, use `COMSIG_ITEM_DROPPED`. ;)

File: code/datums/martial.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/datums/martial.dm]:
```suggestion
	return//if you want this to continue on to throw_impact, then return TRUE
```

**Comment by OverlyGeneric** [comment]
> Every time he drops an item he can't help but kick it.


Welp, now he must wear football gear and be able to impale people with limbs by chucking them really hard.

**Comment by tralezab** [comment]
Football helmet seems really hard to sprite but that is my favorite idea for his costume so far

also progress has been a little slow since i have so many projects gonna try to make some more work on this tonight

**Comment by tralezab** [comment]
maybe returning later, i want to finish fugitives for sure

**Pull request closed without merging**